### ISSUE TITLE
 PHP Fatal error

### ISSUE THREAD

**Issue by lfcunha** [slimphp/Slim#1070]
Upon upgrading from version 2.5.0 to 2.6.2 I get a PHP fatal error. Reverting to version 2.5.0 prevents the error. There's no stack traced generated. Only this message in the Apache error log:

PHP Fatal error:  Call to a member function set() on a non-object in vendor/slim/slim/Slim/Slim.php on line 239, referer: https://cripdb.mssm.edu/login

**Comment by akrabat** [comment]
Can you reproduce this in a small test application?

**Comment by lfcunha** [comment]
I stripped down my application to demo this. Please get it from this repo:
https://github.com/lfcunha/testslim26

(the app is hardcoded to be served from testslim26 directory: www.example.com/testslim26)

If you change slim's version to 2.6.\* in the composer.json file and update, you'll see the error.

**Comment by akrabat** [comment]
The problem is that that you controller extends `Slim\Slim` and you override the constructor. If you want to do this, then you must call through to the parent constructor. 

Similarly, if you want to override `render()`, then it must have the same function signature as the parent class.

So, your `homeController` needs to look like:

``` php
class homeController extends \Slim\slim {

    public function __construct(array $userSettings = array()) {
        parent::__construct($userSettings);
        $this->app = \Slim\Slim::getInstance();
    }

    public function render($template, $data = array(), $status = null) {
        $this->app->render('home.twig', array("user"=>"Luis"));
    }
}
```

However, as you instantiate `homeController` in routes.php, you don't actually need it to extend Slim, so it should probably look like this:

``` php
class homeController {

    public function __construct() {
        $this->app = \Slim\Slim::getInstance();
    }

    public function render() {
        $this->app->render('home.twig', array("user"=>"Luis"));
    }
}
```

This way, `homeController` is no longer an instance of `Slim` itself, but rather holds an instance of `Slim` as `$app`.

So, in summary, Slim is working as expected.

**Comment by lfcunha** [comment]
The render method was actually called something else in my app (I did not mean to override render()). However, just removing "extends \Slim\slim" from the controllers stopped this issue (It was not causing a problem up to version 2.5).
Thank you for your help.
